modules/renter/contractor/editor.go

```diff
@@ -104,9 +104,19 @@ func (he *hostEditor) Close() error {
 }
 
 // Upload negotiates a revision that adds a sector to a file contract.
-func (he *hostEditor) Upload(data []byte) (crypto.Hash, error) {
+func (he *hostEditor) Upload(data []byte) (_ crypto.Hash, err error) {
```

Comment:
Having this change alone makes me nervous, because you are depending on the values of these variables in a `defer` check. But I'm pretty sure if you do something like `return errors.New("an error")` the deferred code will miss the fact that you returned an error. Shadowed variables could also be an issue.

So, all the rest of the return statements in this function need to be updated to just say `return`, I think.